lib/ConstExtract/ConstExtract.cpp

```diff
@@ -119,17 +120,58 @@ parseProtocolListFromFile(StringRef protocolListFilePath,
   return true;
 }
 
+static std::string extractLiteralOutput(Expr *expr) {
+  std::string LiteralOutput;
+  llvm::raw_string_ostream OutputStream(LiteralOutput);
+  expr->printConstExprValue(&OutputStream, nullptr);
+
+  return LiteralOutput;
+}
+
 static std::shared_ptr<CompileTimeValue>
 extractPropertyInitializationValue(VarDecl *propertyDecl) {
   auto binding = propertyDecl->getParentPatternBinding();
   if (binding) {
     auto originalInit = binding->getOriginalInit(0);
     if (originalInit) {
-      std::string LiteralOutput;
-      llvm::raw_string_ostream OutputStream(LiteralOutput);
-      originalInit->printConstExprValue(&OutputStream, nullptr);
-      if (!LiteralOutput.empty())
-        return std::make_shared<RawLiteralValue>(LiteralOutput);
+      auto literalOutput = extractLiteralOutput(originalInit);
+      if (!literalOutput.empty()) {
+        return std::make_shared<RawLiteralValue>(literalOutput);
+      }
+
+      if (auto callExpr = dyn_cast<CallExpr>(originalInit)) {
+        std::vector<FunctionParameter> parameters;
+
+        const auto args = callExpr->getArgs();
+        for (auto arg : *args) {
+          auto label = arg.getLabel().str().str();
+          auto expr = arg.getExpr();
+
+          switch (expr->getKind()) {
+          case ExprKind::DefaultArgument: {
+            auto defaultArgument = cast<DefaultArgumentExpr>(expr);
+            auto *decl = defaultArgument->getParamDecl();
+
+            if (decl->hasDefaultExpr()) {
+              literalOutput =
+                  extractLiteralOutput(decl->getTypeCheckedDefaultExpr());
+            }
+
+            break;
+          }
+          default:
+            literalOutput = extractLiteralOutput(expr);
+            break;
+          }
+
+          if (!literalOutput.empty()) {
```

Comment:
If we failed to extract a literal value for this argument, I think we should still collect an argument as `RuntimeValue`. This way our representation of the collected call will have a matching number of arguments to what the compiler sees. What do you think?

Reply:
If any value cannot be extracted at compile time, it may be best to abort and propagate `RuntimeValue` all the way up to the top-most value (once nesting is supported, such as for dictionaries). Otherwise, any eventual consumers of the JSON might attempt to utilize the data without realizing that it's can't be funny reconstructed at runtime from the provided detail alone.

Reply:
> If any value cannot be extracted at compile time, it may be best to abort and propagate `RuntimeValue` all the way up to the top-most value (once nesting is supported, such as for dictionaries). Otherwise, any eventual consumers of the JSON might attempt to utilize the data without realizing that it's can't be funny reconstructed at runtime from the provided detail alone.

I think it may be best to leave this responsibility to JSON clients. We can imagine use-cases where we wish to extract metadata about properties that are default-initialized with an `init` call where say the first argument is a compile-time constant `String`, and is the only one relevant to build-time tooling, but also has additional arguments whose values are not relevant at build-time and are only known at runtime.

Reply:
@quinntaylor @artemcm - I pushed an update that supports both `RuntimeValue` and `RawLiteralValue` extraction for function parameters. Currently, if raw literal extract fails, we default to a runtime value. Extracting other types can be added as needed.

On the JSON rendering side, I added `valueKind` to the function parameter to differentiate values.

```swift
public struct Foo {
    let p12 = Bat(buz: "hello", fuz: adder(2, 3))
}

func adder(_ x: Int, _ y: Int) -> Int {
    x + y
}

public struct Bat {
    let buz: String
    let fuz: Int

    init(buz: String = "", fuz: Int = 0) {
        self.buz = buz
        self.fuz = fuz
    }
}
```

```json
{
  "label": "p12",
  "type": "fields.Bat",
  "isStatic": "false",
  "isComputed": "false",
  "valueKind": "InitCall",
  "value": {
    "type": "fields.Bat",
    "arguments": [
      {
        "label": "buz",
        "type": "Swift.String",
        "valueKind": "RawLiteral",
        "value": "\"hello\""
      },
      {
        "label": "fuz",
        "type": "Swift.Int",
        "valueKind": "Runtime"
      }
    ]
  }
},
```